crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -0,0 +1,318 @@
+﻿use crate::system::{Command, EntityCommands};
+use crate::{entity::Entity, reflect::ReflectComponent, world::World};
+use bevy_reflect::{Reflect, TypeRegistry};
+
+/// An extension trait for [`EntityCommands`] for reflection related functions
+pub trait EntityCommandsReflectExtension {
```

Comment:
I think the separation makes logical sense and I do like it being separated because at least from a code perspective it keeps all the more special purpose reflection functions away from the general functions which should be much more commonly used. However it could go either way and I don't really have a strong opinion on it. I do think if we added more functions then the division might get more useful just from a code base perspective since the actual EntityCommands file is getting rather big already. It also depends on how "public" reflection related things are supposed to be which I don't have a great gauge on so I'll defer to others on that point.

Additionally I know at least for me I use an IDE and I very rarely purposefully import things. I might be an outlier but I would assume most people use something that will autocomplete and suggest related functions even if they aren't already imported which would make the discoverability point much less damning.

I'd overall be in favor of keeping the extension trait.